addons/controllers/clusterbootstrap_controller.go

```diff
@@ -604,6 +617,39 @@ func (r *ClusterBootstrapReconciler) mergeClusterBootstrapPackagesWithTemplate(
 	return packages, nil
 }
 
+func (r *ClusterBootstrapReconciler) getPackageFromCBTInResolvedTKR(
+	packageRefName string,
+	cluster *clusterapiv1beta1.Cluster,
+	updatedClusterBootstrap *runtanzuv1alpha3.ClusterBootstrap,
+	log logr.Logger) (*runtanzuv1alpha3.ClusterBootstrapPackage, error) {
+
+	clusterBootstrapTemplate := &runtanzuv1alpha3.ClusterBootstrapTemplate{}
+	key := client.ObjectKey{Namespace: r.Config.SystemNamespace, Name: updatedClusterBootstrap.Status.ResolvedTKR}
+	if err := r.Client.Get(r.context, key, clusterBootstrapTemplate); err != nil {
+		log.Error(err, "unable to fetch ClusterBootstrapTemplate", "objectkey", key)
+		return nil, err
+	}
+	for _, pkg := range clusterBootstrapTemplate.Spec.AdditionalPackages {
+		// use the refName in package CR, since the package CR hasn't been cloned at this point, use SystemNamespace to fetch packageCR
+		pkgRefName, _, err := util.GetPackageMetadata(r.context, r.aggregatedAPIResourcesClient, pkg.RefName, r.Config.SystemNamespace)
```

Comment:
This call to GetPackageMetadata is not needed since we just need to compare the packageRefName passed into the function with whats there in clusterBootstrapTemplate 

i.e. 
```
if packageRefName == pkg.RefName {
   return pkg, nil
}
```

This should fix the unit test failure.

Reply:
Fixed